jina/hubble/hubio.py

```diff
@@ -332,7 +341,16 @@ def pull(self) -> str:
                 if scheme == 'jinahub+docker':
                     self._load_docker_client()
                     st.update(f'Pulling {executor.image_name}...')
-                    self._client.images.pull(executor.image_name)
+                    try:
+                        self._client.images.pull(executor.image_name)
+                    except docker.errors.APIError:
+                        img_not_found = False
+                        try:
+                            self._client.images.get(executor.image_name)
+                        except docker.errors.ImageNotFound:
+                            img_not_found = True
+                        if img_not_found:
```

Comment:
this seems weird, why not just raising and avoid this extra boolean?

Reply:
Actually I need to re-raise APIError.
Maybe I can create a helper function `docker_image_exists(image_name)` because the same logic is also used in other places (dockerize.py, container/__init__.py).
I'm worried about the usage of docker (if I'll put it in a helper function) since docker is not installed in all tags.
What do you think ?

Reply:
anyway, u can do except APIError as apiex, and then raise apiex. And u would not need this variable

Reply:
done